codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/serialize/JsonSerializerGenerator.kt

```diff
@@ -337,10 +337,7 @@ class JsonSerializerGenerator(
         val writer = context.writerExpression
         val value = context.valueExpression
         when (target) {
-            is StringShape -> when (target.hasTrait<EnumTrait>()) {
-                true -> rust("$writer.string(${value.name}.as_str());")
-                false -> rust("$writer.string(${value.name});")
-            }
+            is StringShape -> rust("$writer.string(${value.name}.as_str());")
```

Comment:
sneaking suspicion clippy doesn't like this (but we can just add a clippy ignore)

Reply:
I had the same feeling, but SDK tests (and clippy) are not complaining on my local environment. Let's see what the CI have to say.

Reply:
CI is happy 💪🏼

Reply:
Hum you are probably right, it should be called only if the target is an Enum.. I'll give it a try to narrow this down.